src/hausdorff.jl

```diff
@@ -9,7 +9,13 @@ struct MaxReduction  <: ReductionOperation end
 struct MeanReduction <: ReductionOperation end
 
 _reduce(op::MaxReduction, x)  = maximum(x)
-_reduce(op::MeanReduction, x) = sum(x) / length(x)
+function _reduce(op::MeanReduction, x)
+    if length(x) == 0
+        return zero(eltype(x))
+    else
+        return sum(x) / length(x)
+    end
+end
```

Comment:
curious to ask, is this equivalent to `mean(x)`? I'm not very sure if it should be `0` when `length(x) == 0` here.

Reply:
Thanks for noticing and sorry, I think I checked that against `Statistics.mean` but now I checked again and it actually returns `NaN` as a mean of an empty collection. I'll update it.